### PR TITLE
Add Support abortCallback Option

### PR DESCRIPTION
**Pull request by wbw20**
# Add Support for abortCallback Option

## Todo
+ [x] specs
+ [ ] plus one of team

@zendesk/orchid Better version of https://github.com/zendesk/ember-resource/pull/112.  Now we can pass an `abortCallback` into Em.Resource.ajax which will receive the ajax.abort function as an argument so that consumers can do with it what they please in a transparent way.

like so:

```js
var cancel;

Em.Resource.ajax({
  url: '/autocomplete',
  abortCallback: function(abort) {
    cancel = abort;
  }
});

/* some time later */

cancel();
```

Obviously this isn't perfect but @shajith and @vcekov agree it is the most pragmatic step at this point.

### CODE REVIEW

File: dist/ember-resource.js [hunk 1]
```diff
@@ -451,7 +451,7 @@ if (typeof this === 'object') this.LRUCache = LRUCache;
 
     var dfd = $.Deferred();
 
-    $.ajax(options).done(function() {
+    var ajax = $.ajax(options).done(function() {
```
Comment (kencheeto) [inline dist/ember-resource.js]: don't need this anymore, right?

File: spec/javascripts/fetchSpec.js [hunk 2]
```diff
@@ -223,4 +223,51 @@ describe('deferred fetch', function() {
...
+        request.done(done);
+        server.respond();
+        expect(done.called).to.be.true;
+      });
+
+      it('should not call the done callback when abort is called', function() {
+        request = Em.Resource.ajax({
+          url: '/autocomplete',
+          abortCallback: function(abort) {
+            abort();
```
Comment (shajith) [inline spec/javascripts/fetchSpec.js]: do you want to write this in a way where you save off the `abort`, and call it outside of this `ajax` invocation (the way you plan to use this, basically)?

File: dist/ember-resource.js [hunk 3]
```diff
@@ -463,6 +463,10 @@ if (typeof this === 'object') this.LRUCache = LRUCache;
       });
     });
 
+    if (options.abortCallback) {
+      options.abortCallback(ajax.abort.bind(ajax));
+    }
+
     return dfd.promise();
   };
```
Comment (shajith) [inline dist/ember-resource.js]: you can bump the dist file after merging the PR

**Comment by shajith** [comment]
+1, left a note on the spec, and get rid of the dist changes

**Comment by vcekov** [comment]
+1

File: spec/javascripts/fetchSpec.js (hunk 2 above)
Comment (wbw20) [inline spec/javascripts/fetchSpec.js]: The principal says merge!

**Pull request merged**